segmentation/utils/data_utils.py

```diff
@@ -142,20 +156,53 @@ def load_imgs_from_dir(data_dir, img_sub_folder=None, fovs=None, imgs=None,
     if np.min(img_data) < 0:
         raise ValueError("Integer overflow from loading TIF image, try a larger dtype")
 
-    # remove .tif or .tiff from image name
-    img_names = [os.path.splitext(img)[0] for img in imgs]
-
     if variable_sizes:
         row_coords, col_coords = range(1024), range(1024)
     else:
         row_coords, col_coords = range(test_img.shape[0]), range(test_img.shape[1])
 
+    if use_filenames:
+        fovs = [fov.split(delimiter)[0] for fov in fovs]
+        img_names = [image_name]
+
```

Comment:
This sort of goes along with what Noah discussed in the meeting. The logic looks good, but it just might be a bit much to incorporate several different cases into the same function which could become confusing if it becomes a habit. Again, it might still be a good idea to condense all this into one function for now, but as more options become available, it might be best to split into separate functions.

Reply:
It's no trouble to move this into a separate function, so that's what I'll do.  I'll probably rename this current function tho and make the appropriate reference changes in other files+nb's.  While I'm at it, I'll add the separate file-checking function and fully fledge it out when I make a PR for 'in-dockerland path checking'